std/math.d

```diff
@@ -7291,3 +7292,134 @@ unittest
         }
     }
 }
+
+/**
+ * Gives the next power of two after $(D val). $(T) can be any built-in
+ * numerical type.
+ *
+ * Params:
+ *     val = any number
+ *
+ * Returns:
+ *     the next power of two after $(D val)
+ */
+T nextPow2(T)(const T val) if (isIntegral!T)
+{
+    Unqual!T x = val;
+
+    static if(isSigned!T) x = abs(x);
+
+    x |= x >> 1;
+    x |= x >> 2;
+    x |= x >> 4;
+
+    static if (T.sizeof > 1)
+        x |= x >> 8;
+    static if (T.sizeof > 2)
+        x |= x >> 16;
+    static if (T.sizeof > 4)
+        x |= x >> 32;
+
+    if (val < 0)
```

Comment:
>plus copysign from val to x for floating points

`copysign` is already being used in the floating point version, and you're the one who wrote it ;)